### DISCUSSION TITLE
How to do something before next start without custom server

### DISCUSSION THREAD

**Discussion by Capricair** [vercel/next.js#47193] [Help] [answered]
### Summary

How to connect redis before next start, I don't want to use a custom server, because the official document says that a custom server will delete important performance optimizations

**Comment by icyJoseph** [comment, marked as answer]
Are you hosting on Vercel? If so, your app is deployed as a serverless app. So wherever you need the db connection, you make one, if one doesn't exist already. Do make sure to kill it though, but after some timeout the serverless handler is terminated. Making a few simplifications for sake of argument.

If you are hosting yourself, maybe in next.config.js, there's a thread on this forum discussing this very topic: https://github.com/vercel/next.js/discussions/15341 with a ton of tricks from the community.

**Reply by Capricair** [reply]
next.config.js phase can do this, thanks!
